### ISSUE TITLE
stringifying circular structure can lead to no report in bugsnag

### ISSUE THREAD

**Issue by rokerkony** [bugsnag/bugsnag-angular#3]
Hi,
I tried to use this lib but when I throw an error in some bigger component meaning with more dependencies which have their own deps then I get
```ts
[bugsnag] TypeError: Converting circular structure to JSON
    at Object.stringify (<anonymous>)
    at stringify (bugsnag.js:796)
    at _$payload_20 (bugsnag.js:1756)
...
```

in other words, a report is not sent afterward because it is not even able to stringify it...

It would be nice to have all the info about a component but it is very dangerous as it can break a code and not send a report to bugsnag

the problematic part is:
https://github.com/bugsnag/bugsnag-angular/blob/master/src/index.ts#L24
and context as well

**Comment by bengourley** [comment]
Hmm. Are you able to make a reduced test case? The bugsnag library is omits circular references in circular data structures using [fast safe stringify](https://github.com/davidmarkclements/fast-safe-stringify).

Also sorry for the delay in getting to your issue. I missed it over the holiday season!

**Comment by bengourley** [comment]
I found a way to reproduce this. If an object implements a `toJSON` method and it returns a circular structure, it results in this error:

```js
var obj = {}

obj.toJSON = function () {
  var a = {}
  a.a = a
  return a
}

obj.toJSON.forceDecirc = true

bugsnagClient.notify(new Error('Circ'), {
  metaData: { obj: obj }
})
```

We'll look into a fix for it.

**Comment by bengourley** [comment]
This should now be fixed as of [bugsnag-js v4.6.0](https://github.com/bugsnag/bugsnag-js/releases/tag/v4.6.0). Please upgrade!
